src/main/java/com/example/bookshop/repository/book/BookRepository.java

```diff
@@ -1,9 +1,17 @@
 package com.example.bookshop.repository.book;
 
 import com.example.bookshop.model.Book;
+import java.util.List;
 import org.springframework.data.jpa.repository.JpaRepository;
 import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+import org.springframework.stereotype.Repository;
 
+@Repository
 public interface BookRepository extends JpaRepository<Book, Long>, JpaSpecificationExecutor<Book> {
 
+    @Query("SELECT b FROM Book b JOIN b.categories c WHERE c.id = :categoryId "
+            + "AND b.isDeleted = false")
```

Comment:
```suggestion
    @Query("SELECT b FROM Book b JOIN b.categories c WHERE c.id = :categoryId ")
```
Do you really need it?